packages/nuxt/src/core/app.ts

```diff
@@ -141,6 +139,14 @@ async function resolveApp (nuxt: Nuxt, app: NuxtApp) {
     ].map(plugin => normalizePlugin(plugin as NuxtPlugin)))
   }
 
+  // Add back plugins not specified in layers or user config
+  for (const p of nuxt.options.plugins) {
+    const plugin = normalizePlugin(p)
+    if (!app.plugins.some(p => p.src === plugin.src)) {
+      app.plugins.unshift(plugin)
```

Comment:
Maybe this should be `push` instead of `unshift`? (https://github.com/nuxt/nuxt/issues/23163)